### PR TITLE
Build boost with flag -fPIC 

### PR DESCRIPTION
**Pull request by liliajsonar**
Build boost with flag -fPIC so that it can be used in dynamic libraries (Derek)

### CODE REVIEW

File: SonarExternal.cmake [hunk 1]
```diff
@@ -1280,7 +1280,7 @@ function(build_boost)
       --with-libraries=${WITH_LIBRARIES}
     BUILD_IN_SOURCE YES
     BUILD_COMMAND ./b2 -j8
-    INSTALL_COMMAND ./b2 install
+    INSTALL_COMMAND ./bjam cxxflags=-fPIC cflags=-fPIC -a install
```
Comment (yhager) [inline SonarExternal.cmake]: what's the `-a` for?

Comment (yhagerjs) [reply SonarExternal.cmake]: `bjam` is deprecated, please stay with `b2`.

Comment (yhagerjs) [reply SonarExternal.cmake]: also, do you really need `cflags`? I don't think boost has any C components

**Comment by yhagerjs** [comment]
I've merged a similar fix instead of this one.

**Comment by liliajsonar** [comment]
Thanks Yuval!

**Pull request closed without merging**